jsre/pretty.go

```diff
@@ -0,0 +1,181 @@
+package jsre
+
+import (
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/robertkrimen/otto"
+)
+
+var (
+	functionColor = color.New(color.FgMagenta)
+	specialColor  = color.New(color.Bold)
+	numberColor   = color.New(color.FgRed)
+	stringColor   = color.New(color.FgGreen)
+)
+
+type ppctx struct{ vm *otto.Otto }
+
+func prettyPrint(vm *otto.Otto, value otto.Value) {
+	ppctx{vm}.printValue(value, "")
+}
+
+func (ctx ppctx) printValue(v otto.Value, indent string) {
+	switch {
+	case v.IsObject():
+		ctx.printObject(v.Object(), indent)
+	case v.IsNull():
+		specialColor.Print("null")
+	case v.IsUndefined():
+		specialColor.Print("undefined")
+	case v.IsString():
+		s, _ := v.ToString()
+		stringColor.Printf("%q", s)
+	case v.IsBoolean():
+		b, _ := v.ToBoolean()
+		specialColor.Printf("%t", b)
+	case v.IsNaN():
+		numberColor.Printf("NaN")
+	case v.IsNumber():
+		s, _ := v.ToString()
+		numberColor.Printf("%s", s)
+	default:
+		fmt.Printf("%s<unprintable>", indent)
+	}
+}
+
+func (ctx ppctx) printObject(obj *otto.Object, indent string) {
```

Comment:
Is there a reason why you're using this type of "type casting" over `obj.Value()` and use [`value.Export`](https://github.com/robertkrimen/otto#func-value-export)?